django/db/models/query.py

```diff
@@ -1566,12 +1566,13 @@ def prefetch_one_level(instances, prefetcher, lookup, level):
     #  callable that gets value to be matched for returned instances,
     #  callable that gets value to be matched for passed in instances,
     #  boolean that is True for singly related objects,
-    #  cache name to assign to).
+    #  cache or field name to assign to,
+    #  boolean that is True when the previous argument is a cache name vs a field name).
 
     # The 'values to be matched' must be hashable as they will be used
     # in a dictionary.
 
-    rel_qs, rel_obj_attr, instance_attr, single, cache_name = (
+    rel_qs, rel_obj_attr, instance_attr, single, cache_name, is_cache = (
```

Comment:
This would be clearer (to me) if the sense were reversed and it were named `is_descriptor` instead of `is_cache`. Would that naming be wrong in some way that I'm missing?

I find this naming confusing because it's named `is_cache` but isn't really describing whether anything is (or is not) "a cache." An alternative to `is_descriptor` if you want to keep the same sense would be something like `is_cache_safe` -- but I still think `is_descriptor` is clearer.